frontend/src/components/ConfirmDialog.tsx

```diff
@@ -1,36 +1,44 @@
-// ConfirmDialog.tsx  - material ui
+import CloseIcon from '@mui/icons-material/Close';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
 import Dialog from '@mui/material/Dialog';
-import DialogTitle from '@mui/material/DialogTitle';
-import DialogContent from '@mui/material/DialogContent';
 import DialogActions from '@mui/material/DialogActions';
-import Button from '@mui/material/Button';
-import Box from '@mui/material/Box';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
 import IconButton from '@mui/material/IconButton';
 import Typography from '@mui/material/Typography';
-import CloseIcon from '@mui/material/IconButton';
+import React from 'react';
 
-const ConfirmDialog = (message: String) => {
-    return (
-        <Dialog open={true} maxWidth="sm" fullWidth>
-            <DialogTitle>Confirm the action</DialogTitle>
-            <Box position="absolute" top={0} right={0}>
-                <IconButton>
-                    <CloseIcon />
-                </IconButton>
-            </Box>
-            <DialogContent>
-                <Typography>{message}</Typography>
-            </DialogContent>
-            <DialogActions>
-                <Button color="primary" variant="contained">
-                    Cancel
-                </Button>
-                <Button color="secondary" variant="contained">
-                    Confirm
-                </Button>
-            </DialogActions>
-        </Dialog>
-    );
-};
+interface Props {
+  message: string;
+  onCancel: React.MouseEventHandler<HTMLButtonElement>;
+  onConfirm: React.MouseEventHandler<HTMLButtonElement>;
```

Comment:
I've tried to use this type before and usually end up taking it out, so this type will probably need to be changed later to something closer to `VoidFunction` or `(event?) => void`. I figure we can change it when trying to use the `ConfirmDialog`.